sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -592,7 +592,18 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
   private def convertToStructField(col: QualifiedColType): StructField = {
     val builder = new MetadataBuilder
     col.comment.foreach(builder.putString("comment", _))
-    StructField(col.name.head, col.dataType, nullable = true, builder.build())
+    col.default.foreach{ value: String =>
+      builder.putString(DefaultCols.CURRENT_DEFAULT_COLUMN_METADATA_KEY, value)
+    }
+    val result = StructField(col.name.head, col.dataType, nullable = true, builder.build())
+    if (col.default.isDefined) {
+      lazy val analyzer = new Analyzer(catalogManager)
```

Comment:
Shall we perform the constant folding in `AlterTableAddColumnsCommand`?
So that we can avoid creating a new Analyzer within `ResolveSessionCatalog`, which is actually part of the Analyzer.

Reply:
Good idea, done. Now we can continue using the same `Analyzer`.